tests/test_e2e_80_pathfinder.py

```diff
@@ -0,0 +1,160 @@
+    def teardown_class(cls):
+        cls.net.stop()
+
+    def restart(self, _clean_config=False, _enable_all=False):
+
+        # Start the controller setting an environment in which the setting is
+        # preserved (persistence) and avoid the default enabling of all elements
+        self.net.start_controller(clean_config=_clean_config, enable_all=_enable_all)
+        self.net.wait_switches_connect()
+
+        # Wait a few seconds to kytos execute LLDP
+        time.sleep(10)
+    
+    def add_topology_metadata(self):
+        """Add topology metadata."""
+        links_metadata = {
+            "74bbc9527a0e309a86c95744042bcf9e3beb52955c942cac5fc735b1cf986f7f": {
+                "link_name": "s1-eth3-s6-eth3",
+                "ownership": "red",
+                "bandwidth": "10",
+                "delay" : 100,
+                "priority": 120,
+            },
+            "cf0f4071be426b3f745027f5d22bc61f8312ae86293c9b28e7e66015607a9260": {
+                "link_name": "s1-eth2-s2-eth2",
+                "ownership": "blue",
+                "bandwidth": "100",
+                "delay" : 10,
+                "priority": 5,                
+            },
+            "adda3859b963110d584bf6ec3ac85ddea80276001e37edc1c420463a34c80c9e": {
+                "link_name": "s2-eth4-s6-eth4",
+                "ownership": "blue",
+                "bandwidth": "100",
+                "delay" : 10,
+                "priority": 5,                
+            },
+            
+        }
+
+        for link_id, metadata in links_metadata.items():
+            api_url = f"{KYTOS_API}/topology/v3/links/{link_id}/metadata"
+            response = requests.post(
+                api_url,
+                data=json.dumps(metadata),
+                headers={"Content-type": "application/json"},
+            )
+            assert response.status_code == 201, response.text
+        return links_metadata        
+
+    @pytest.mark.parametrize("undesired_link, expected", 
+                            [([], 1),(["74bbc9527a0e309a86c95744042bcf9e3beb52955c942cac5fc735b1cf986f7f"], 8)])
+    def test_10_undesired_link_and_max_path(self, undesired_link, expected):
+        """Tests fastest path from switch 1 to 6 then blocks it"""
+        api_url = KYTOS_API + '/pathfinder/v3/'
+        post_body = {
+"source": "00:00:00:00:00:00:00:01:3",
+"destination": "00:00:00:00:00:00:00:06:3",
+"undesired_links": undesired_link,
+"spf_attribute": "hop",
+"spf_max_paths": 1,
+"parameter": "hop"
+}
+
+        response = requests.post(api_url, json=post_body)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert data['paths'][0]['cost'] == expected, f'Shortest path expected {expected}: {data}'
+
+    @pytest.mark.parametrize("attribute, expected_cost", 
+                            [("hop", 1), ("delay", 20), ("priority", 15)])
+    def test_20_spf_attribute_and_max_path_cost(self, attribute, expected_cost):
```

Comment:
Can you refactor `expected_cost` with a more descriptive name for this context, something like `expected_max_cost`? Also, same as prior comments, lets assert which paths are expected for completeness.